src/ui/simulator/toolbox/components/datagrid/renderer/area/renewable.areasummary.cpp

```diff
@@ -0,0 +1,256 @@
+    // gp : do we wish to have the line empty if cluster disabled
+    // if (!cluster->enabled)
+    //    return wxEmptyString;
+    switch (x)
+    {
+    case 0:
+        return wxStringFromUTF8(cluster->group());
+    case 1:
+        return cluster->enabled ? wxT("Yes") : wxT("no");
+    case 2:
+        return wxString() << cluster->unitCount;
+    case 3:
+        return DoubleToWxString(cluster->nominalCapacity);
+    }
+    return wxEmptyString;
+}
+
+double RenewableClusterSummarySingleArea::cellNumericValue(int x, int y) const
+{
+    Data::RenewableCluster* cluster = (pArea and (uint) y < pArea->renewable.list.size())
+                                      ? pArea->renewable.list.byIndex[y]
+                                      : nullptr;
+    // gp : do we wish to have the line empty if cluster disabled
+    // if (!cluster->enabled)
+    //     return 0.;
+    switch (x)
+    {
+    case 0:
+        return 0.;
+    case 1:
+        return cluster->enabled ? 1. : 0.;
+    case 2:
+        return cluster->unitCount;
+    case 3:
+        return cluster->nominalCapacity;
+    }
+    return 0.;
+}
+
+void RenewableClusterSummarySingleArea::onAreaChanged(Antares::Data::Area* area)
+{
+    if (pArea != area)
+    {
+        pArea = area;
+        RefreshAllControls(pControl);
+    }
+}
+
+IRenderer::CellStyle RenewableClusterSummarySingleArea::cellStyle(int col, int row) const
+{
+    return (col > 0 and Math::Zero(cellNumericValue(col, row)))
+             ? IRenderer::cellStyleDefaultDisabled
+             : (col == 1 || col == 2) ? IRenderer::cellStyleConstraintWeight
+                                      : IRenderer::cellStyleDefault;
+}
+
+struct NoCheck
+{
+    template<class T>
+    static bool Validate(const T&)
+    {
+        return true;
+    }
+};
+
+struct CheckUnitCount
+{
+    static bool Validate(uint& f)
+    {
+        if (f > 100)
+            f = 100;
+        return true;
+    }
+};
+
+template<class CheckT>
+static bool UpdateUnsignedLong(uint& value, const String& str)
+{
+    uint l;
+    if (str.to(l))
+    {
+        if (value != l and CheckT::Validate(l))
+        {
+            value = l;
+            MarkTheStudyAsModified();
+            OnInspectorRefresh(nullptr);
+            return true;
+        }
+    }
+    return false;
+}
+
+template<class CheckT>
+static bool UpdateDouble(double& value, const String& str)
+{
+    double d;
+    if (str.to(d))
+    {
+        if (not Math::Equals<double>(value, d))
+        {
+            if (CheckT::Validate(d))
+            {
+                value = d;
+                OnInspectorRefresh(nullptr);
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+template<class CheckT>
+static bool UpdateBool(bool& value, const String& str)
+{
+    bool d;
+    if (str.to(d))
+    {
+        if (value != d)
+        {
+            if (CheckT::Validate(d))
+            {
+                value = d;
```

Comment:
Same remark as above : use templates and move to generic class. Then use where needed (thermal, renewable, elsewhere,  etc.).